src/Dapr.Jobs.Analyzer/DaprJobsAnalyzerAnalyzer.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Dapr.Jobs.Analyzer
+{
+    /// <summary>
+    /// DaprJobsAnalyzerAnalyzer.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DaprJobsAnalyzerAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor DaprJobHandlerRule = new DiagnosticDescriptor(
+            id: "DAPRJOBS0001",
+            title: "Ensure Post Mapper handler is present for all the Scheduled Jobs",
+            messageFormat: "There should be a mapping post endpoint for each scheduled job to make sure app receives notifications for all the scheduled jobs",
```

Comment:
Could we reword this to be something like "Job invocations require the `MapJobInvocationHandler` be set and configured for each anticipated job on `IEndpointRouteBuilder`" with something similar for the title?